# Pheme dataset

Let's first see what kinds of events inside the Pheme dataset.

Note that we have a .DS_Store, we need to drop that.

In [1]:
ls

Preprocessing Pheme dataset.ipynb  convert_veracity_annotations.py
all-rnr-annotated-threads/


In [4]:
import os

dirs = []
for dirt in os.listdir('all-rnr-annotated-threads'):
    if dirt != '.DS_Store':
        dirs.append(dirt)

print(dirs)

['germanwings-crash-all-rnr-threads', 'prince-toronto-all-rnr-threads', 'ebola-essien-all-rnr-threads', 'ferguson-all-rnr-threads', 'charliehebdo-all-rnr-threads', 'putinmissing-all-rnr-threads', 'ottawashooting-all-rnr-threads', 'sydneysiege-all-rnr-threads', 'gurlitt-all-rnr-threads']


The sturcture of one event is like below:

- non-rumours
- rumours
- .DS_Store

Try to use one first, and next is just a bigger `for` loop.

In [5]:
# use 
dirs[0]

'germanwings-crash-all-rnr-threads'

In [48]:
a = []
a.append('dsada')
a

['dsada']

In [49]:
import json
import pandas as pd

In [63]:
'''
dirct ---

- non-rumours
- rumours
'''
data = []
path = 'all-rnr-annotated-threads/{}'
for di in dirs:
    for dirct in os.listdir(path.format(di)):
        if dirct != '.DS_Store':
            for num_dir in os.listdir((path + '/{}').format(di, dirct)):
                # now in one dir inside the non-rumors
                if num_dir != '.DS_Store':
#                     print(num_dir)
                    '''
                    - source-tweets 
                    - reactions


                    one data sample: x_source + x_reactions
                    '''
                    one_row_sample = []
                    one_data_sample = ''
                    for d in os.listdir((path + '/{}/{}').format(di, dirct, num_dir)):  
                        if d == 'source-tweets':
                            for j_file in os.listdir((path + '/{}/{}/{}').format(di, dirct, num_dir, d)):
                                if j_file != '.DS_Store':
                                    j_file_path = (path + '/{}/{}/{}/{}').format(di, dirct, num_dir, d, j_file)
                                    j = json.load(open(j_file_path, 'rb'))
                                    source = j['text'] # this is a str
                                    one_data_sample = one_data_sample + source

                        if d == 'reactions':
                            for j_file in os.listdir((path + '/{}/{}/{}').format(di, dirct, num_dir, d)):
                                if j_file != '.DS_Store':
                                    j_file_path = (path + '/{}/{}/{}/{}').format(di, dirct, num_dir, d, j_file)
                                    j = json.load(open(j_file_path, 'rb'))
                                    reaction = j['text'] # str
                                    one_data_sample = one_data_sample + reaction

                    one_row_sample.append(one_data_sample)
                    if dirct == 'non-rumours':
                        one_row_sample.append('non-rumours')
                    if dirct == 'rumours':
                        one_row_sample.append('rumours')
                    
                    data.append(one_row_sample)

df = pd.DataFrame(data, columns=['Text', 'Label'])
df

Text        Label
0     #BREAKING: A Germanwings Airbus A320 has crash...      rumours
1     Updated numbers @AP: BREAKING: #Germanwings CE...      rumours
2     @YanniKouts @germanwings @flightradar24 A bit ...      rumours
3     #Germanwings co-pilot suffered serious depress...      rumours
4     @ThisIsGaZa إِنَّا لِلّهِ وَإِنَّـا إِلَيْهِ ر...      rumours
...                                                 ...          ...
6420  Franz Marc: Horses... Update-#Gurlitt Nazi-Tai...  non-rumours
6421  Munich District Court has confirmed the applic...  non-rumours
6422  Where should the Gurlitt collection go?  :Many...  non-rumours
6423  @DrDonnaYates They should insure the Swiss art...  non-rumours
6424  ‘The Gurlitt collection should be sold to bene...  non-rumours

[6425 rows x 2 columns]

In [64]:
df.to_csv('Pheme.csv')

Did a correct preprocessing, 

Rumours + non-Rumours total get 6425.

通过向Paper [Link](https://arxiv.org/pdf/1806.03713v1.pdf) 求证， 我是对的！